code/modules/admin/topic.dm

```diff
@@ -11,30 +11,48 @@
 		return TRUE
 	log_admin_private("[key_name(usr)] clicked an href with [msg] authorization key! [href]")
 
-/datum/admins/Topic(href, href_list)
-	..()
+// BASE ADMINS_TOPIC
+
+/datum/datum_topic/admins_topic
+	var/log = TRUE
+	var/key_valid
+
+/datum/datum_topic/admins_topic/proc/TryRun(list/input)
+	. = Run(input)
+	if(islist(.))
+		. = list2params(.)
+
+/datum/datum_topic/admins_topic/proc/Run(list/input)
+	CRASH("Run() not implemented for [type]!")
+
+
+// ADMINS TOPIC()
 
+/datum/admins/Topic(href,href_list)
+	..()
 	if(usr.client != src.owner || !check_rights(0))
 		message_admins("[usr.key] has attempted to override the admin panel!")
 		log_admin("[key_name(usr)] tried to use the admin panel without authorization.")
 		return
 
 	if(!CheckAdminHref(href, href_list))
 		return
-
-	if(href_list["ahelp"])
-		if(!check_rights(R_ADMIN, TRUE))
-			return
-
-		var/ahelp_ref = href_list["ahelp"]
-		var/datum/admin_help/AH = locate(ahelp_ref)
-		if(AH)
-			AH.Action(href_list["ahelp_action"])
-		else
-			to_chat(usr, "Ticket [ahelp_ref] has been deleted!")
-
-	else if(href_list["ahelp_tickets"])
-		GLOB.ahelp_tickets.BrowseTickets(text2num(href_list["ahelp_tickets"]))
+	var/list/input = null
+	var/static/list/topic_handlers = TopicHandlers(/datum/datum_topic/admins_topic)
+
+	input = href_list
+	var/datum/datum_topic/admins_topic/handler
+	for(var/I in topic_handlers)
+		if(I in input)
+			handler = topic_handlers[I]
+			break
+	/*
+	if((!handler || initial(handler.log)) && config && CONFIG_GET(flag/log_admin))

```

Comment:
I am not sure if I need this or how I would change it to properly handle the logging.